instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/JarServiceNameDetector.java

```diff
@@ -129,7 +129,8 @@ private static String getServiceName(Path jarPath) {
 
   @Override
   public int order() {
-    // make it run later than the SpringBootServiceNameDetector
-    return 1000;
+    // make it run earlier than the SpringBootServiceNameDetector
```

Comment:
My understanding is that currently the service name detector that runs earlier sets the name. Service detectors that run later check whether the name is already set, and if it is, just don't do anything. Given that logic running `JarServiceNameDetector` before `SpringBootServiceNameDetector` doesn't make sense. Have I misunderstood something?

Reply:
I thought that all of our detectors use `shouldApply`, could you point out which detectors don't?

Reply:
Most detectors do not use `shouldApply`, e.g. [OsResourceDetector](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/6b607c1e18e6936f7a7eef46a3be77892cef9b3a/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/OsResourceProvider.java#L17)

Reply:
Service name detectors are special, they have to use `shouldApply` and bail out if the service name is already set. Other detectors don't need to do this.